torch/csrc/jit/codegen/cuda/index_compute.cpp

```diff
@@ -815,25 +815,35 @@ kir::TensorIndex* Index::getGlobalProducerIndex(
   }
 
   kir::Val* cur_stride = ir_builder.create<kir::Int>(1);
-  for (size_t i = 0; i < root_dom.size(); i++) {
-    auto dim = root_dom.size() - i - 1;
-    if (root_dom[dim]->isReduction()) {
-      continue;
-    }
-    if (root_dom[dim]->isBroadcast()) {
-      continue;
-    }
-    if (producer_tv->domain()->contiguity()[dim]) {
-      strides[dim] = cur_stride;
-      cur_stride = ir_builder.mulExpr(
-          cur_stride, gpu_lower->lowerValue(root_dom[dim]->extent()));
-    } else {
-      cur_stride = strides[dim];
+  // if we have rfactor we can't simplify the indexing like this, we would need
+  // to fix contiguity size to be rfactor size not root size
+  if (root_dom.size() == producer_tv->domain()->contiguity().size()) {
+    for (size_t i = 0; i < root_dom.size(); i++) {
+      auto dim = root_dom.size() - i - 1;
+      if (root_dom[dim]->isReduction()) {
+        continue;
+      }
+      if (root_dom[dim]->isBroadcast()) {
+        continue;
+      }
+
+      kir::Val* root_ind = nullptr;
+      auto kir_root_dom =
+          gpu_lower->lowerValue(root_dom[dim])->as<kir::IterDomain>();
+      if (producer_indexing.indexMap().find(kir_root_dom) !=
+          producer_indexing.indexMap().end()) {
+        root_ind = producer_indexing.indexMap().at(kir_root_dom);
+      }
+
+      if (producer_tv->domain()->contiguity()[dim] && root_ind->isZeroInt()) {
```

Comment:
Why contiguity is only honored when `root_ind` is zero?

Reply:
If we index into a 3D tensor with a 1D index because it's contiguous and we're doing linear indexing the index will be placed at the left most position, but we need the right most stride. I.E. `T0[bidx * gdimx + tidx]` will look to us at this point in 3D as `T0[bidx * gdimx + tidx, 0, 0]` but strides will be `T0[stride[0], stride[1], stride[2]]` and we're trying to get the left most index aligned with the right most stride.

Reply:
So what I'm trying to do is slide the stride information left when I detect it's a contiguous indexing.